app/src/main/java/ch/sdp/vibester/model/OfflineSongList.kt

```diff
@@ -0,0 +1,129 @@
+package ch.sdp.vibester.model
+
+import android.content.Context
+import android.os.Environment
+import ch.sdp.vibester.api.LastfmMethod
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileReader
+
+/**
+ * Process the fetched data from the external storage of the app.
+ * Mainly, it creates a list of songs in the form Pair("$songName", "$artistName")
+ * @param ctx: Context of the caller, to fetch the downloaded folder
+ * @param method: Chosen playlist type
+ */
+class OfflineSongList(ctx: Context, method: String) {
+    private var songList = mutableListOf<Pair<String, String>>()
+    private var page = ""
+    private var songsPerPage = ""
+    private var totalPages = ""
+    private var totalSongs = ""
+    private var context = ctx
+    private var emptySongs: Boolean = false
+
+    init {
+        try {
+            var tracksField = "tracks"
+            if (method == LastfmMethod.BY_ARTIST.method) {
+                tracksField = "toptracks"
+            }
+
+            fillList()
+
+            page = "1"
+            songsPerPage = "100"
+            totalPages = "20"
+            totalSongs = "2000"
+
+        } catch (e: Exception) {
+            throw IllegalArgumentException("OfflineSongList constructor, failed reading from file or empty file!")
+        }
+    }
+
+    /**
+     * Adds from the downloads to the list of songs ("$songName", "$artistName")
+     * Saves the list of songs in songList
+     */
+    private fun fillList() {
+        var records = File(context.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "records.txt")
+
+
+        if(!records.exists() || records.length() == 0L) {
```

Comment:
Add a space for readability `if (!records.exists() || records.length() == 0L)`

Reply:
Done